java/ql/src/Metrics/Summaries/GeneratedVsManualCoverage.ql

```diff
@@ -0,0 +1,64 @@
+/**
+ * @id java/summary/generated-vs-manual-coverage
+ * @name Metrics of generated versus manual MaD coverage
+ * @description Expose metrics for the number of API endpoints covered by generated versus manual MaD models.
+ * @kind table
+ * @tags summary
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSummary
+import utils.modelgenerator.internal.CaptureModels
+
+/**
+ * Returns the number of `DataFlowTargetApi`s with Summary MaD models
+ * for a given package and provenance.
+ */
+bindingset[package]
+private float getNumMadModeledApis(string package, string provenance) {
+  provenance in ["generated", "manual", "both"] and
+  result =
+    count(SummarizedCallable sc |
+      package = sc.asCallable().getCompilationUnit().getPackage().getName() and
+      exists(DataFlowTargetApi dataFlowTargApi | sc.asCallable() = dataFlowTargApi) and
+      (
+        sc.isAutoGenerated() and // "auto-only"
+        provenance = "generated"
+        or
+        sc.isManuallyGenerated() and // "manual-only"
+        provenance = "manual"
+        or
+        sc.isBothAutoAndManuallyGenerated() and // "both"
+        provenance = "both"
+      )
+    )
+}
+
+/** Returns the total number of `DataFlowTargetApi`s for a given package. */
+bindingset[package]
+private float getNumApis(string package) {
+  result =
+    count(DataFlowTargetApi dataFlowTargApi |
+      package = dataFlowTargApi.getCompilationUnit().getPackage().getName()
+    )
+}
+
+from
+  DataFlowTargetApi dataFlowTargApi, string package, float generatedOnly, float both,
+  float manualOnly, float non, float all, float generatedCoverage, float manualCoverage
+where
+  // bind `package` to a `DataFlowTargetApi` package name
+  package = dataFlowTargApi.getCompilationUnit().getPackage().getName() and
+  // count the number of APIs with generated-only, both, and manual-only MaD models for each package
+  generatedOnly = getNumMadModeledApis(package, "generated") and
+  both = getNumMadModeledApis(package, "both") and
+  manualOnly = getNumMadModeledApis(package, "manual") and
+  // count the total number of `DataFlowTargetApi`s for each package
+  all = getNumApis(package) and
+  non = all - (generatedOnly + both + manualOnly) and
+  // Proportion of manual models covered by generated ones
+  generatedCoverage = (both / (both + manualOnly)) and
+  // Proportion of generated models covered by manual ones
+  manualCoverage = (both / (both + generatedOnly))
+select package, generatedOnly, both, manualOnly, non, all, generatedCoverage, manualCoverage
```

Comment:
Th first 5 numbers here are integers, why not just keep their types?

Reply:
I had them as floats for the sake of avoiding integer division in the metrics calculations. For some reason when I had tried to cast to float before it didn't work (I think I must have mis-typed something). I've now switched them to ints and put a cast to float on the metrics calculations in https://github.com/github/codeql/pull/11585/commits/e42d9e4930704217f139963495f19c71363e66a1.